### PR TITLE
wifi_mamager/wm_test: make the "security mode" of join command is not necessary; 

### PR DESCRIPTION
**Pull request by glb1013**
user can join the network even if he does not know about the exactly security mode of the network.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [6482ebc071396a36614fa5cd809ca378cac87ecd] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [6482ebc071396a36614fa5cd809ca378cac87ecd] - Code Rule Check (C++) OK.

File: framework/src/wifi_manager/slsi_wifi_utils.c [hunk 1]
```diff
@@ -354,6 +407,27 @@ wifi_utils_result_e wifi_utils_connect_ap(wifi_utils_ap_config_s *ap_connect_con
 	slsi_security_config_t *config = NULL;
 
 	if (ap_connect_config->passphrase_length > 0) {
+		// scan to get the security config if it is unknown
```
Comment (SeonghoByeon) [inline framework/src/wifi_manager/slsi_wifi_utils.c]: One question from my side;
do we really need (or manage) the additional scanning in this layer?
I guess the slsi WiFi driver might have a process that finds the auth and crypto within the driver layer.
Please verify that feature exists in the driver code or not.

**Comment by SeonghoByeon** [comment]
Hi @glb1013 , now I am waiting from your comment. This patch is indeed required for better support of wifi join/scan. I have plan to finish this work within this August. Please give any feedback, or we can take over together. Thanks.

**Comment by zhouxinhe** [comment]
> Hi @glb1013 , now I am waiting from your comment. This patch is indeed required for better support of wifi join/scan. I have plan to finish this work within this August. Please give any feedback, or we can take over together. Thanks.

Hi @SeonghoByeon , I will handle your comments since now, maybe I will raise new PR later.

**Comment by zhouxinhe** [comment]
> > Hi @glb1013 , now I am waiting from your comment. This patch is indeed required for better support of wifi join/scan. I have plan to finish this work within this August. Please give any feedback, or we can take over together. Thanks.
> 
> Hi @SeonghoByeon , I will handle your comments since now, maybe I will raise new PR later.

I raised new PR: #3683, please review, thanks.

File: apps/examples/wifi_manager_sample/wm_test.c [hunk 2]
```diff
@@ -1305,31 +1305,37 @@ int wm_parse_commands(struct options *opt, int argc, char *argv[])
...
 			} else {
-				return -1;
+				opt->crypto_type = WIFI_MANAGER_CRYPTO_UNKNOWN;
 			}
 		} else {
-			opt->crypto_type = get_crypto_type(argv[4]);
+			opt->crypto_type = get_crypto_type(argv[5]);
 			if (opt->crypto_type == WIFI_MANAGER_CRYPTO_UNKNOWN) {
-				return -1;
+				opt->auth_type = WIFI_MANAGER_AUTH_UNKNOWN;
```
[the remark this answers is not in the document]
Comment (zhouxinhe) [reply apps/examples/wifi_manager_sample/wm_test.c]: I reverted it(and other 2 places above) in new PR, thanks.

File: apps/examples/wifi_manager_sample/wm_test.c [hunk 3]
```diff
@@ -1305,31 +1305,37 @@ int wm_parse_commands(struct options *opt, int argc, char *argv[])
 		}
 		opt->func = wm_connect;
 		opt->ssid = argv[3];
-		opt->auth_type = get_auth_type(argv[4]);
+		opt->password = argv[4];
```
[the remark this answers is not in the document]
Comment (zhouxinhe) [reply apps/examples/wifi_manager_sample/wm_test.c]: fixed in new PR:
e.g. valid usage as below:
wm_test join test-ap wpa2_aes 12345678  // argc=6
wm_test join test-ap 12345678  // argc=5, unspecified security mode
wm_test join test-ap open          // argc=5, open mode without password

File: framework/src/wifi_manager/slsi_wifi_utils.c [hunk 4]
```diff
@@ -354,6 +407,27 @@ wifi_utils_result_e wifi_utils_connect_ap(wifi_utils_ap_config_s *ap_connect_con
...
+		if (ap_connect_config->ap_auth_type == WIFI_UTILS_AUTH_UNKNOWN ||
+			ap_connect_config->ap_crypto_type == WIFI_UTILS_CRYPTO_UNKNOWN) {
+			wuret = wifi_utils_scan_ap(ap_connect_config);
+			if (wuret != WIFI_UTILS_SUCCESS) {
+				return wuret;
+			}
+
+			if (scan_filter_result.result_list != NULL) {
+				// Use the auth_type/crypto_type value of first one.
```
[the remark this answers is not in the document]
Comment (zhouxinhe) [reply framework/src/wifi_manager/slsi_wifi_utils.c]: fixed in new PR, thanks.

File: framework/src/wifi_manager/slsi_wifi_utils.c (hunk 1 above)
Comment (zhouxinhe) [reply framework/src/wifi_manager/slsi_wifi_utils.c]: I checked slsi WiFi driver, there's no such feature to scan specific ssid. 
I also checked @glb1013's modification in driver, it cannot make any optimization, so I removed his change.

**Comment by xixidodo** [comment]
#3683 merged   @SeonghoByeon    Please close this one

**Pull request closed without merging**